wavefront_python_sdk/common/utils.py

```diff
@@ -0,0 +1,241 @@
+        str_builder.append("source=" + sanitize(source))
+        if tags is not None:
+            for key in tags:
+                if is_blank(key):
+                    raise ValueError("Histogram tag key cannot be blank")
+                if is_blank(tags[key]):
+                    raise ValueError("Histogram tag value cannot be blank")
+                str_builder.append(sanitize(key) + '=' + sanitize(tags[key]))
+        line_builder.append(' '.join('{0}'.format(s) for s in str_builder))
+    return '\n'.join('{0}'.format(line) for line in line_builder) + "\n"
+
+
+def tracing_span_to_line_data(name, start_millis, duration_millis, source,
+                              trace_id, span_id, parents, follows_from, tags,
+                              span_logs, default_source):
+    """
+    Wavefront Tracing Span Data format
+    <tracingSpanName> source=<source> [pointTags] <start_millis>
+    <duration_milli_seconds>
+    Example: "getAllUsers source=localhost
+              traceId=7b3bf470-9456-11e8-9eb6-529269fb1459
+              spanId=0313bafe-9457-11e8-9eb6-529269fb1459
+              parent=2f64e538-9457-11e8-9eb6-529269fb1459
+              application=Wavefront http.method=GET
+              1533531013 343500"
+    @param name: Span Name
+    @type name: str
+    @param start_millis: Start time
+    @type start_millis: long
+    @param duration_millis: Duration time
+    @type duration_millis: long
+    @param source: Source
+    @type source: str
+    @param trace_id: Trace ID
+    @type trace_id: UUID
+    @param span_id: Span ID
+    @type span_id: UUID
+    @param parents: Parents Span ID
+    @type parents: List of UUID
+    @param follows_from: Follows Span ID
+    @type follows_from: List of UUID
+    @param tags: Tags
+    @type tags: list
+    @param span_logs: Span Log
+    @param default_source: Default Source
+    @type default_source: str
+    @return: String data of tracing span
+    """
+    if is_blank(name):
+        raise ValueError("Span name cannot be blank")
+
+    if is_blank(source):
+        source = default_source
+
+    str_builder = [sanitize(name),
+                   "source=" + sanitize(source),
+                   "traceId=" + str(trace_id),
+                   "spanId=" + str(span_id)]
+    if parents is not None:
+        for uuid in parents:
+            str_builder.append("parent=" + str(uuid))
+    if follows_from is not None:
+        for uuid in follows_from:
+            str_builder.append("followsFrom=" + str(uuid))
+    if tags is not None:
+        for key, val in tags:
+            if is_blank(key):
+                raise ValueError("Span tag key cannot be blank")
+            if is_blank(val):
+                raise ValueError("Span tag val cannot be blank")
+            str_builder.append(sanitize(key) + "=" + sanitize(val))
+    str_builder.append(start_millis)
+    str_builder.append(duration_millis)
+    return ' '.join('{0}'.format(s) for s in str_builder) + "\n"
```

Comment:
`'{0}'.format()` here is useless and expensive, if all the items in `str_builder` are strings.

Reply:
I'd be `' '.join(str_builder) + '\n'`

Reply:
No, it's not all strings. Can't do this.

Reply:
Please make sure these are all strings when appending to `str_builder`.